### PR TITLE
Mw hrastega patch 1

### PR DESCRIPTION
**Pull request by mw-hrastega**
- Documented the build tool support for UI and pipeline workflows.
- Updated screenshots.
- Made enhancements wherever possible.

@Reviewers: Please perform a detailed review as a lot of things have changed.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Continuous Integration with MATLAB on Jenkins
-This plugin enables you to run MATLAB&reg; scripts, functions, and statements as part of your build. You also can run MATLAB and Simulink&reg; tests, and generate test and coverage artifacts such as JUnit-style test results and Cobertura code and model coverage reports.
+This plugin enables you to build and test your MATLAB&reg; project as part of your Jenkins&trade; build. For example, you can automatically identify any code issues in your project, run tests and generate test and coverage artifacts, and package your files into a toolbox.
```
Comment (mcafaro) [inline README.md:2]: Are we misleading by saying "you can automatically identify any code issues .... package your files into a toolbox"? Our plugin doesn't really assist you with those workflows in particular.

File: CONFIGDOC.md [hunk 2]
```diff
@@ -42,89 +44,104 @@ You can use the [`matlabroot`](https://www.mathworks.com/help/matlab/ref/matlabr
...
-When you set up the **Build** section of the project configuration window, the plugin provides you with the **Run MATLAB Command** and **Run MATLAB Tests** build steps.
-
-If you use a source code management (SCM) system such as Git&trade;, then your project should include the appropriate SCM configuration to check out the code before it can invoke the plugin. If you do not use any SCM systems to manage your code, then an additional build step might be required to ensure that the code is available in the project workspace before the build starts.
+When you set up the **Build Steps** section of the project configuration window, the plugin provides you with three build steps:
 
-#### Run MATLAB Command
-The **Run MATLAB Command** build step lets you run MATLAB scripts, functions, and statements. You can use this build step to customize your test run or execute any MATLAB commands.
+* To run a MATLAB build, use the [Run MATLAB Build](#run-matlab-build) step.
+* To run MATLAB and Simulink tests and generate artifacts, use the [Run MATLAB Tests](#run-matlab-tests) step.
+* To run a MATLAB script, function, or statement, use the [Run MATLAB Command](#run-matlab-command) step.
```
Comment (mcafaro) [inline CONFIGDOC.md:51]: At some point (not necessarily now), it feels like we should add a table or something explaining when you would want to use one of these steps vs the others.

- Run MATLAB Build: For sophisticated workflows that require more than just running tests
- Run MATLAB Tests: If you only care about running tests 
- Run MATLAB Command: For simple workflows that just require running a command

File: CONFIGDOC.md [hunk 3]
```diff
@@ -204,68 +227,66 @@ node {
...
-Use the `runMATLABCommand` step in your Pipeline to run MATLAB scripts, functions, and statements. You can use this step to customize your test run or execute any MATLAB commands.
+### Use the `runMATLABBuild` Step
+Use the `runMATLABBuild` step in your pipeline to run a build using the [MATLAB build tool](https://www.mathworks.com/help/matlab/matlab_prog/overview-of-matlab-build-tool.html). You can use this step to run the tasks specified in a file named `buildfile.m` in the root of your repository. To use the `runMATLABBuild` step, you need MATLAB R2022b or a later release.
 
-You must provide `runMATLABCommand` with a string that specifies the script, function, or statement you want to execute. If you specify more than one script, function, or statement, use a comma or semicolon to separate them. If you want to run a script or function, do not specify the file extension.
+The `runMATLABBuild` step accepts an optional input. Use this input to specify the tasks to run. If you specify the step in your `Jenkinsfile` without an input (`runMATLABBuild()`), the plugin runs the default tasks in `buildfile.m` as well as all the tasks on which they depend.
```
Comment (mcafaro) [inline CONFIGDOC.md:233]: `runMATLABBuild()` will be fixed when this doc goes out?

File: CONFIGDOC.md [hunk 4]
```diff
@@ -204,68 +227,66 @@ node {
...
 
-You must provide `runMATLABCommand` with a string that specifies the script, function, or statement you want to execute. If you specify more than one script, function, or statement, use a comma or semicolon to separate them. If you want to run a script or function, do not specify the file extension.
+The `runMATLABBuild` step accepts an optional input. Use this input to specify the tasks to run. If you specify the step in your `Jenkinsfile` without an input (`runMATLABBuild()`), the plugin runs the default tasks in `buildfile.m` as well as all the tasks on which they depend.
 
-**Example:** `runMATLABCommand 'myscript'`<br/>
-**Example:** `runMATLABCommand 'results = runtests, assertSuccess(results);'` 
+Input                     | Description    
+------------------------- | ---------------
+`tasks`                   | (Optional) Task to execute. If you specify more than one task, use a space to separate them.<br/>**Example:** `'mytask'`<br/>**Example:** `'compile' 'test'`
```
Comment (mcafaro) [inline CONFIGDOC.md:237]:
```suggestion
`tasks`                   | (Optional) Task to execute. If you specify more than one task, use a space to separate them.<br/>**Example:** `'mytask'`<br/>**Example:** `'compile test'`

```

File: CONFIGDOC.md [hunk 5]
```diff
@@ -354,7 +413,7 @@ pipeline {
                             runMATLABCommand 'pwd'
```
Comment (mcafaro) [inline CONFIGDOC.md:413]: Why are these 2 separate runMATLABCommand steps? Why not just `runMATLABCommand 'ver'` or `runMATLABCommand 'ver, pwd'`?

File: CONFIGDOC.md [hunk 6]
```diff
@@ -460,8 +513,7 @@ pipeline {
                         }
                     }
                     stage('Run MATLAB Tests') {
-                    steps
-                        {
+                    steps {
```
Comment (mcafaro) [inline CONFIGDOC.md:516]: Indentation looks off here

File: CONFIGDOC.md (hunk 2 above)
Comment (mw-hrastega) [reply CONFIGDOC.md:51]: Like! Good suggestion.

File: CONFIGDOC.md (hunk 3 above)
Comment (mw-hrastega) [reply CONFIGDOC.md:233]: Yes, @davidbuzinski said it will be a simple fix.

File: CONFIGDOC.md (hunk 4 above)
Comment (mw-hrastega) [reply CONFIGDOC.md:237]: Good catch. Fixed.

File: CONFIGDOC.md (hunk 5 above)
Comment (mw-hrastega) [reply CONFIGDOC.md]: Updated as suggested.

File: CONFIGDOC.md (hunk 6 above)
Comment (mw-hrastega) [reply CONFIGDOC.md]: Fixed.

File: README.md (hunk 1 above)
Comment (mw-hrastega) [reply README.md:2]: I am not sure—even if one doesn't use the "Run MATLAB Build" step, the simple "Run MATLAB Command" step makes it possible to run a script and achieve all these objectives.

Comment (mcafaro) [reply README.md:2]: True, I'm just not sure the plugin itself is contributing to those goals much.

File: CONFIGDOC.md (hunk 3 above)
Comment (davidbuzinski) [reply CONFIGDOC.md:233]: Yes, in the Jenkinsfile step syntax the `tasks` option needs to be switched from required to optional. I'll make sure to submit this change before we release.

**Review by davidbuzinski — APPROVED** [review]

**Pull request merged**